### ISSUE TITLE
Non-idempotency in canonical SMILES

### ISSUE THREAD

**Issue by brilee** [rdkit/rdkit#4596]
**Describe the bug**
O\N=C1\C(\Nc2ccccc\12)=C1\C(=O)Nc2c1cccc2Br breaks idempotency in canonical SMILES generation.

**To Reproduce**
```python
def f(s):
  return Chem.MolToSmiles(Chem.MolFromSmiles(s), isomericSmiles=True, canonical=True)

bad_string = r'O\N=C1\C(\Nc2ccccc\12)=C1\C(=O)Nc2c1cccc2Br'
print(bad_string)
print(f(bad_string))
print(f(f(bad_string)))

O\N=C1\C(\Nc2ccccc\12)=C1\C(=O)Nc2c1cccc2Br
O=C1Nc2c(Br)cccc2/C1=C1/Nc2ccccc2/C1=N\O
O=C1Nc2c(Br)cccc2/C1=C1/Nc2ccccc2/C1=N/O
```
**Expected behavior**
f(f(s)) == f(s) should be true for a canonicalization algorithm.

**Configuration (please complete the following information):**
 - RDKit version: rdkit-pypi-2021.3.5.1
 - OS: Google Colab?
 - Python version (if relevant): 3.7.12
 - Are you using conda? No
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit? !pip install rdkit-pypi

**Comment by IchiruTake** [comment]
> example is not even eventually

I suppose that it is not a good idea to call canonical and isomeric at a time. Normally, you can prevent this by simplify the SMILES to achieve good read. Then pass to canonicallize or isomeric it from your request.
You can attempt this by first, generate a SMILES without cis-trans identifier `/` and/or `\`.  Then you insert the cis-trans identifier by your request. A good note is that verify every input. Build good input first and justify them instead of calling.
Related to your problem, I suppose that is not actually cycle, this actually the string behaviour. I have tested your string and found that the mol is flipped relied on 'one' path of the ring. In particularly, I have benchmark your string to see your actual result (see below)

```
smi=c1ccccc1/C=N\O
smi=c1ccccc1/C=N/O
smi=O=C1Nc2c(Br)cccc2/C1=C1/Nc2ccccc2/C1=N\O
smi=O=C1Nc2c(Br)cccc2/C1=C1/Nc2ccccc2/C1=N/O
smi=C/C=C1/Nc2ccccc2/C1=N\O
smi=C/C=C1/Nc2ccccc2/C1=N/O
smi=C\C=C1/Nc2ccccc2/C1=N\O
smi=C\C=C1/Nc2ccccc2/C1=N/O
```
In particularly, four last string can recall your problem which could be the main issue for this. I am not so confident about your result but there is my assumption related. However, since the first two SMILES cannot recall your problem because there are no cis-trans identifier aforehead. When the first cis-trans identifier called and can be created, the cis-trans path found and since the ring can make two path to the next cis-trans double bond, it can flip your next smiles. 

In conclusion, this behaviour is not actually the bug but it is the Smiles problem as cyclo-hexane can also recall your problem. This behaviour should be modified either by RDKit implementation to guarantee consistency.
